crates/conductor_lib/src/holochain.rs

```diff
@@ -240,53 +239,37 @@ impl Holochain {
     }
 
     fn call_inner(
-        me: &Holochain,
+        context: Arc<Context>,
         zome: &str,
         cap: CapabilityRequest,
         fn_name: &str,
         params: &str,
     ) -> HolochainResult<JsonString> {
-        me.check_instance()?;
-        me.check_active()?;
-
         let zome_call = ZomeFnCall::new(&zome, cap, &fn_name, JsonString::from_json(&params));
-        let context = me.context()?;
         Ok(context.block_on(call_zome_function(zome_call, context.clone()))?)
     }
 
     /// call a function in a zome
-    pub fn call(
-        &self,
+    pub fn call_zome_function(
+        context: Arc<Context>,
         zome: &str,
         cap: CapabilityRequest,
         fn_name: &str,
         params: &str,
     ) -> HolochainResult<JsonString> {
-        let context = self.context.as_ref().unwrap();
         let metric_name = format!("{}.{}", zome, fn_name);
         with_latency_publishing!(
             metric_name,
             context.metric_publisher,
             Self::call_inner,
-            self,
+            context.clone(),
```

Comment:
a field of the context and the context itself are both passed into this macro, could be unified

Reply:
The 3rd macro argument is a function to call and all following arguments are passed to that function. So the cloned context is actually the first argument to `Self::call_inner`.

Reply:
@lucksus I'm talking about the second and fourth arguments, `context.metric_publisher` and `context`

Reply:
Oh I see, it's kind of a funky macro. I guess this works.